src/test/java/io/github/ust/mico/kafkafaasconnector/MessageListenerTests.java

```diff
@@ -191,5 +270,75 @@ public void testRouteHistory() {
+        String testFilterTopic = "TestFilterTopic";
+        cloudEventSimple.setFilterOutBeforeTopic(testFilterTopic);
+        cloudEventSimple.setIsTestMessage(true);
+
+        assertTrue("The message should be filtered out", cloudEventSimple.isFilterOutNecessary(testFilterTopic));
+    }
+
+    @Test
+    public void testNotFilterOutCheck() {
+        MicoCloudEventImpl<JsonNode> cloudEventSimple = TestConstants.basicCloudEventWithRandomId();
+        String testFilterTopic = "TestFilterTopic";
+        cloudEventSimple.setFilterOutBeforeTopic(testFilterTopic);
+
+        assertFalse("The message not should be filtered out, because it is not a test message", cloudEventSimple.isFilterOutNecessary(testFilterTopic));
+    }
 
+    @Test
+    public void testNotFilterOutCheckDifferentTopics() {
+        MicoCloudEventImpl<JsonNode> cloudEventSimple = TestConstants.basicCloudEventWithRandomId();
+        String testFilterTopic = "TestFilterTopic";
+        cloudEventSimple.setFilterOutBeforeTopic(testFilterTopic);
+        cloudEventSimple.setIsTestMessage(true);
+
+        assertFalse("The message not should be filtered out, because it has not reached the filter out topic", cloudEventSimple.isFilterOutNecessary(testFilterTopic + "Difference"));
+    }
+
+    @Test
+    public void testNotFilterNormalMessages() {
+        EmbeddedKafkaBroker embeddedKafka = broker.getEmbeddedKafka();
+        Consumer<String, MicoCloudEventImpl<JsonNode>> consumer = TestConstants.getKafkaConsumer(embeddedKafka);
+        embeddedKafka.consumeFromAnEmbeddedTopic(consumer, kafkaConfig.getOutputTopic());
+        KafkaTemplate<String, MicoCloudEventImpl<JsonNode>> template = TestConstants.getKafkaProducer(embeddedKafka);
+
+        MicoCloudEventImpl<JsonNode> cloudEventSimple = TestConstants.basicCloudEventWithRandomId();
+        template.send(kafkaConfig.getInputTopic(), "0", cloudEventSimple);
+
+        ConsumerRecord<String, MicoCloudEventImpl<JsonNode>> event = KafkaTestUtils.getSingleRecord(consumer, kafkaConfig.getOutputTopic(), 1000);
+        assertThat(event, is(notNullValue()));
+        assertThat(event.value().getId(), is(equalTo(cloudEventSimple.getId())));
+
+        // Don't forget to detach the consumer from kafka!
+        consumer.unsubscribe();
+        consumer.close();
+    }
+
+    @Test
+    public void testFilterTestMessages() {
+        EmbeddedKafkaBroker embeddedKafka = broker.getEmbeddedKafka();
+        Consumer<String, MicoCloudEventImpl<JsonNode>> consumer = TestConstants.getKafkaConsumer(embeddedKafka);
+        embeddedKafka.consumeFromEmbeddedTopics(consumer, kafkaConfig.getFilteredTestMessagesTopic(), kafkaConfig.getOutputTopic());
+        KafkaTemplate<String, MicoCloudEventImpl<JsonNode>> template = TestConstants.getKafkaProducer(embeddedKafka);
+
+        MicoCloudEventImpl<JsonNode> cloudEventSimple = TestConstants.basicCloudEventWithRandomId();
+        cloudEventSimple.setIsTestMessage(true);
```

Comment:
Could use convenience method in test constants

Reply:
In some of my tests I need isTestMessage and FilterOutBeforeTopic and in some only one of them. I wanted to make clear that this test needs both and the convenience method hides this. And it only one line shorter.